src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -0,0 +1,93 @@
+import { PAGES } from '../../../../content-strings';
+import { Request, Response, CompanyHouseResponse } from '../../../../../types';
+import { TEMPLATES, ROUTES, FIELD_IDS } from '../../../../constants';
+import corePageVariables from '../../../../helpers/page-variables/core/insurance';
+import api from '../../../../api';
+import companiesHouseValidation from './validation/companies-house';
+import companyHouseResponseValidation from './validation/companies-house-response';
+
+import { companyHouseSummaryList } from '../../../../helpers/summary-lists/company-house-summary-list';
+
+const { COMPANY_HOUSE } = FIELD_IDS.INSURANCE.EXPORTER_BUSINESS;
+
+const get = async (req: Request, res: Response) => {
+  return res.render(TEMPLATES.INSURANCE.EXPORTER_BUSINESS.COMPANY_DETAILS, {
+    ...corePageVariables({
+      PAGE_CONTENT_STRINGS: PAGES.INSURANCE.EXPORTER_BUSINESS.COMPANY_DETAILS,
+      BACK_LINK: req.headers.referer,
+    }),
+    POSTROUTE: ROUTES.INSURANCE.EXPORTER_BUSINESS.COMPANY_HOUSE_SEARCH,
+    FIELD_IDS: COMPANY_HOUSE,
+  });
+};
+
+/**
+ * posts to company house api
+ * returns summary list if company found
+ * or returns validation errors if not found, errors with input or companies house api is down
+ */
```

Comment:
Could we use `@param` etc? Started using this in other PR https://github.com/UK-Export-Finance/exip/blob/main-application/e2e-tests/cypress/e2e/journeys/insurance/eligibility/exporter-location/exporter-location.spec.js

Reply:
done